src/ray/util/event.cc

```diff
@@ -312,6 +367,33 @@ void RayEvent::SendMessage(const std::string &message) {
     event.mutable_custom_fields()->insert(mp.begin(), mp.end());
 
     EventManager::Instance().Publish(event, custom_fields_);
+  } else if (is_export_event) {
+    std::string event_id_buffer = std::string(kEventIDSize, ' ');
+    FillRandom(&event_id_buffer);
+    event_id = StringToHex(event_id_buffer);
+
+    rpc::ExportEvent export_event;
+    export_event.set_event_id(event_id);
+
+    std::string source_type_as_str = Event_SourceType_Name(context.GetSourceType());
+    rpc::ExportEvent_SourceType source_type_ele;
+    if (rpc::ExportEvent_SourceType_Parse(source_type_as_str, &source_type_ele)) {
+      export_event.set_source_type(source_type_ele);
+    } else {
+      throw std::invalid_argument("Invalid source_type type: " + source_type_as_str);
+    }
+    export_event.set_timestamp(current_sys_time_s());
+
+    std::string event_data_type_name = export_event_data_ptr_->GetTypeName();
+    if (event_data_type_name == "ray.rpc.ExportTaskEventData") {
+      rpc::ExportTaskEventData *task_event_data_ptr =
+          dynamic_cast<rpc::ExportTaskEventData *>(export_event_data_ptr_.get());
+      export_event.mutable_task_event_data()->CopyFrom(*task_event_data_ptr);
+    } else {
+      throw std::invalid_argument("Invalid event_data type: " + event_data_type_name);
```

Comment:
do not throw exception. Ray doesn't handle exception at all. can you just use RAY_LOG(FATAL)?